src/main/java/com/mitchej123/hodgepodge/mixins/early/minecraft/MixinItemEnderPearl.java

```diff
@@ -0,0 +1,32 @@
+package com.mitchej123.hodgepodge.mixins.early.minecraft;
+
+import net.minecraft.entity.item.EntityEnderPearl;
+import net.minecraft.entity.player.EntityPlayer;
+import net.minecraft.item.Item;
+import net.minecraft.item.ItemEnderPearl;
+import net.minecraft.item.ItemStack;
+import net.minecraft.world.World;
+
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.Overwrite;
+
+@Mixin(ItemEnderPearl.class)
+public abstract class MixinItemEnderPearl extends Item {
+
+    /**
+     * @author Colen
+     * @reason Enables enderpearls to be used in creative by removing check.
+     */
+    @Overwrite
+    public ItemStack onItemRightClick(ItemStack itemStackIn, World worldIn, EntityPlayer player) {
```

Comment:
there's no need to have this as an overwrite. Redirecting the call to player.capabilities.isCreativeMode to always return false is cleaner as that way other mixins can still modify the same method

Reply:
I have no idea how one would redirect this specific call within another function. Is there an example to look at?

Reply:
[Here](https://github.com/GTNewHorizons/Hodgepodge/blob/master/src/main/java/com/mitchej123/hodgepodge/mixins/early/minecraft/MixinEntityLivingDrop.java#L29) is an example for PUTFIELD, in your case you need GETFIELD but the rest is identical (except that you need a different target and method of course)

Reply:
@mist475 I think I've done this now, can you double check for me? I haven't done a mixin in this manner before.

Reply:
While that works, that might have side effects. What you should do is:

```java
    @Redirect(
            method = "onItemRightClick",
            at = @At(
                    value = "FIELD",
                    target = "Lnet/minecraft/entity/player/PlayerCapabilities;isCreativeMode:Z"))
    public boolean hodgepodge$removeCreativeCHeck(PlayerCapabilities instance) {
        return false;
    }
```